internal-packages/workflow-designer-ui/src/settings/github-integration/github-integration-setting-form.tsx

```diff
@@ -42,9 +78,72 @@ function Installed({
 }: { repositories: GitHubIntegrationRepository[] }) {
 	const { data: workspace } = useWorkflowDesigner();
 	const { isLoading, data, handleSubmit } = useGitHubIntegrationSetting();
+	const [selectedTrigger, setSelectedTrigger] = useState<
+		WorkspaceGitHubIntegrationTrigger | undefined
+	>(data?.event);
+	const [callsign, setCallsign] = useState(data?.callsign || "");
+	const [selectedNextAction, setSelectedNextAction] = useState<
+		WorkspaceGitHubNextIntegrationAction | undefined
+	>(data?.nextAction);
+	const [payloadMaps, setPayloadMaps] = useState<
+		WorkspaceGitHubIntegrationPayloadNodeMap[]
+	>(data?.payloadMaps || []);
+
+	useEffect(() => {
+		if (data) {
+			setSelectedTrigger(data.event);
+			setCallsign(data.callsign || "");
+			setSelectedNextAction(data.nextAction);
+			setPayloadMaps(data.payloadMaps || []);
+		}
+	}, [data]);
+
 	if (isLoading) {
 		return null;
 	}
+
+	const getAvailablePayloadFields = (
+		trigger: WorkspaceGitHubIntegrationTrigger,
+	) => {
+		const fields = Object.values(WorkspaceGitHubIntegrationPayloadField.Enum);
+		const triggerParts = trigger.split(".");
+		const triggerPrefix = `${triggerParts[0]}.${triggerParts[1]}`;
+		return fields.filter((field) => field.startsWith(triggerPrefix));
+	};
```

Comment:
Consider moving this definition outside of the component. It might improve organization/readability